lambdas/SaveSnippetToS3/lambda_function.py

```diff
@@ -7,8 +7,15 @@
 import string
 import boto3 # AWS SDK for Python
 
-s3     = boto3.client('s3', 'us-west-2')
+s3 = boto3.client('s3', 'us-west-2')
 client = boto3.client('lambda')
+s3_resource = boto3.resource('s3')
+
+# Returns time the snippet was last modified
+def get_last_modified(bucket, user_id, snippet_key):
```

Comment:
You are making a request to aws to store your object and then another to retrieve it. See if you can do both at once.